parsely/parselyandroid/ParselyMetadata.java

```diff
@@ -0,0 +1,81 @@
+package com.parsely.parselyandroid;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.HashMap;
+import java.util.Map;
+
+/*! \brief Represents post metadata to be passed to Parsely tracking.
+ *
+ *  This class is used to attach a metadata block to a Parse.ly pageview
+ *  request. Pageview metadata is only required for URLs not accessible over the
+ *  internet (i.e. app-only content) or if the customer is using an "in-pixel" integration.
+ *  Otherwise, metadata will be gathered by Parse.ly's crawling infrastructure.
+ */
+public class ParselyMetadata {
+    public ArrayList<String> authors, tags;
+    public String canonicalUrl, section, thumbUrl, title;
+    public Calendar pubDate;
+
+    /* \brief Create a new ParselyMetadata object.
+     *
+     * @param authors         List of authors for the post.
+     * @param canonicalUrl    Canonical URL of the post.
+     * @param section         Section of the post.
+     * @param tags            List of tags for the post.
+     * @param thumbUrl        URL of a thumbnail for the post.
+     * @param title           Title of the post.
+     * @param pubDate         Publish date of the post.
+     */
+    public ParselyMetadata(
+            @Nullable ArrayList<String> authors,
+            @Nullable String canonicalUrl,
+            @Nullable String section,
+            @Nullable ArrayList<String> tags,
+            @Nullable String thumbUrl,
+            @Nullable String title,
+            @Nullable Calendar pubDate
+    ) {
+        this.authors = authors;
+        this.canonicalUrl = canonicalUrl;
+        this.section = section;
+        this.tags = tags;
+        this.thumbUrl = thumbUrl;
+        this.title = title;
+        this.pubDate = pubDate;
+    }
+
+    /* \brief Turn this object into a Map
+     *
+     * @return a Map object representing the metadata.
+     */
+    public Map<String, Object> toMap() {
+        Map<String, Object> output = new HashMap<>();
+        if (this.authors != null) {
+            output.put("authors", this.authors);
+        }
+        if (this.canonicalUrl != null) {
+            output.put("canonical_url", this.canonicalUrl);
```

Comment:
This should actually be `link` on the wire. I'm not positive `canonical_url` won't work in our backend, but I know that `link` will.

Reply:
Oh, weird, I really thought this was `canonical_url`. Thanks for the catch!

Reply:
@emmett9001 is this only with respect to video data, or does the same go for all in-pixel metadata?

Reply:
If this is always mapped to `link`, should I change all the names from canonical url to `link`?

Reply:
It applies to all metadata, not just video. What really matters is the name on the wire being `link` - in other places, it's not super confusing to use `canonical_url` and `link` interchangeably. The ideal situation would indeed be to just use `link` everywhere.

Reply:
I'm going to change it to link everywhere for consistency (except video metadata where the parameter name will be video id)